library/Zend/Mvc/Controller/AbstractRestfulController.php

```diff
@@ -374,8 +375,12 @@ public function onDispatch(MvcEvent $e)
                     $id = null;
                 }
                 $action = 'head';
-                $this->head($id);
+                $headResult = $this->head($id);
                 $response = $e->getResponse();
+                if ($headResult instanceof HttpResponse) {
+                    $response->setStatusCode($headResult->getStatusCode());
```

Comment:
There is a better solution (probably): 
```php
$response = clone $headResult; $response->setContent(''); return $response;
```
then setting that new `$response` in the `MvcEvent`. Not sure why we'd keep re-using the `MvcEvent#getResponse()` as the blueprint where to take our data from.